pg-extend/src/pg_datum.rs

```diff
@@ -110,6 +110,43 @@ impl From<i64> for PgDatum {
     }
 }
 
+impl TryFromPgDatum for String {
+    fn try_from(datum: PgDatum) -> Result<Self, &'static str> {
+        use std::ffi::CStr;
+        use std::os::raw::c_char;
+
+        if let Some(datum) = datum.0 {
+            let text_val = datum as *const pg_sys::text;
+
+            let cstr = unsafe {
+                let val: *mut c_char = pg_sys::text_to_cstring(text_val);
+                CStr::from_ptr(val)
+            };
+
+            match cstr.to_str() {
+                Ok(s) => Ok(s.to_owned()),
+                Err(_) => Err("datum was not valid utf8")
+            }
+        } else {
+            Err("datum was NULL")
+        }
+    }
+}
+
+impl From<String> for PgDatum {
+    fn from(value: String) -> Self {
+        use std::ffi::CString;
+        use std::os::raw::c_char;
+
+        let cstr = CString::new(value).expect("This shouldn't fail");
+        let ptr: *const c_char = cstr.as_ptr();
+
+        let text = unsafe { pg_sys::cstring_to_text(ptr) };
```

Comment:
Does this copy the string? It must be because I think you'd get an sigsegv after the CString is freed.

Reply:
I'm pretty new to FFI and I haven't written any C before so this is where my knowledge falls down.

[`cstring_to_text()`](https://github.com/postgres/postgres/blob/master/src/backend/utils/adt/varlena.c#L152) performs it's own memory allocation so I'm assuming that's why it works. the Rust `CString` will be dropped when it goes out of scope, so this one should be relatively safe as far as I can see.

Reply:
yes, I think this is good. Thanks!